storage/imxrt-flash/flashdrv.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX RT Flash driver
+ *
+ * Copyright 2019 Phoenix Systems
+ * Author: Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "flashdrv.h"
+#include "rom_api.h"
+
+
+static int flash_getProperties(flash_context_t* context)
+{
+	switch (context->address) {
+		case FLEXSPI_DATA_ADDRESS :
+			/* TO DO */
+			break;
+
+		case FLEXSPI2_DATA_ADDRESS :
+			context->instance = 1;
+			context->option.option0.U = 0xC0000008;
+
+			context->properties.size = 0x400000;
+			context->properties.page_size = 0x100;
+			context->properties.sector_size = 0x1000;
+
+			context->buff = malloc(context->properties.sector_size);
+
+			if (context->buff == NULL)
+				return -EFAULT;
+
+			break;
+
+		default:
+			return -ENODEV;
+	}
+
+	return EOK;
+}
+
+
+static int flash_isValidAddress(flash_context_t* context, uint32_t addr, size_t size)
```

Comment:
This kind of functions (isSomething) are better to return 1 for true and 0 for false. This way you can easily use them in *if*.